brave/brave-ios
# Fix #6352: Multichain Portfolio

## Summary of Changes
- Refactored `NetworkSelectionView` into `NetworkSelectionRootView` and add new `NetworkFilterView` wrapper. This was done so we can pass a `Binding<NetworkFilter>` to the view and use it for filtering networks (hit some issues with SwiftUI Bindings when used in `ObservableObject` and syncing between `CryptoStore` / `NetworkStore` / `PortfolioStore`). 
- Show assets from all networks on Portfolio
- Remove network picker from Portfolio (all assets now shown)

This pull request fixes #6352

## Submitter Checklist:


## Test Plan:
1. Open Portfolio
2. Verify total balance displays correctly, should display total balance for all assets on all non-test networks (All Networks filter selected)
3. Verify assets from all networks are displayed in Portfolio (test networks are excluded from 'All Networks')
4. Tap 'All Networks' filter button
5. Verify 'All Networks' row is displayed and selected
6. Select a specific network
7. Verify total balance displays correctly
8. Verify assets from selected network filter are displayed
9. Tap network filter button
10. Verify selected network is displayed as selected


## Screenshots:

![IMG_2B5C2777B524-1](https://user-images.githubusercontent.com/5314553/203353959-830734fa-ec61-40bd-aa28-df786008e0b1.jpeg) | ![IMG_9F8F63E2FF04-1](https://user-images.githubusercontent.com/5314553/203354159-91a75a0e-ea96-41eb-b6dd-3885b1d012eb.jpeg)
--|--

https://user-images.githubusercontent.com/5314553/203352407-68e94684-900b-4322-a7a4-c92988533dc6.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
